src/WebJobs.Script.WebHost/Metrics/LinuxContainerMetricsPublisher.cs

```diff
@@ -0,0 +1,227 @@
+            {
+                FunctionName = functionName,
+                InvocationId = invocationId,
+                Concurrency = concurrency,
+                ExecutionStage = executionStage,
+                IsSucceeded = success,
+                ExecutionTimeSpanInMs = executionTimeSpan,
+                EventTimeStamp = eventTimeStamp,
+                Tenant = _environment.GetEnvironmentVariable(EnvironmentSettingNames.WebSiteStampDeploymentId)?.ToLowerInvariant()
+            };
+
+            if (!_functionActivities.TryAdd(activity))
+            {
+                _logTraceEvent(LogLevel.Warning, string.Format("LinuxContainerMetricsPublisher: Buffer full. Dropping current batch of function activities"));
+                DrainActivities(_currentFunctionActivities, _functionActivities);
+            }
+
+            _logTraceEvent(LogLevel.Information, string.Format("LinuxContainerMetricsPublisher: Added function activity : {0} {1} {2} {3} {4} {5}", functionName, invocationId, concurrency, executionStage, success, executionTimeSpan));
+        }
+
+        public void AddMemoryActivity(DateTime timeStampUtc, long data)
+        {
+            var memoryActivity = new MemoryActivity
+            {
+                CommitSizeInBytes = data,
+                EventTimeStamp = timeStampUtc,
+                Tenant = _environment.GetEnvironmentVariable(EnvironmentSettingNames.WebSiteStampDeploymentId)?.ToLowerInvariant()
+            };
+
+            if (!_memoryActivities.TryAdd(memoryActivity))
+            {
+                _logTraceEvent(LogLevel.Warning, string.Format("LinuxContainerMetricsPublisher: Buffer full. Dropping current batch of memory activities"));
+                DrainActivities(_currentMemoryActivities, _memoryActivities);
+            }
+        }
+
+        private void DrainActivities<T>(ConcurrentQueue<T> currentActivites, BlockingCollection<T> activitiesToProcess)
+        {
+            currentActivites.Clear();
+
+            if (activitiesToProcess.Any())
+            {
+                while (activitiesToProcess.TryTake(out T item))
+                {
+                    currentActivites.Enqueue(item);
+                }
+            }
+        }
+
+        internal void OnFunctionMetricsPublishTimer(object state)
+        {
+            PublishActivity(_currentMemoryActivities, _memoryActivities, PublishMemoryActivityPath);
+            PublishActivity(_currentFunctionActivities, _functionActivities, PublishFunctionActivityPath);
+        }
+
+        private void PublishActivity<T>(ConcurrentQueue<T> currentActivities, BlockingCollection<T> activitiesToProcess, string publishPath)
+        {
+            try
+            {
+                if (currentActivities.Any())
+                {
+                    SendRequest(currentActivities, publishPath);
+                }
+                DrainActivities(currentActivities, activitiesToProcess);
+            }
+            catch (Exception ex)
```

Comment:
Can you use the IsFatal extension method here to avoid swallowing fatal exceptions?